web_widget_image_paint/static/src/js/web_paint.js

```diff
@@ -0,0 +1,607 @@
+/*
+# Copyright 2009-2019 Noviat.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+*/
+odoo.define('web_widget_image_paint.web_paint', function (require) {
+    "use strict";
+    var core = require('web.core');
+    var FormView = require('web.FormView');
+    var BasicFields= require('web.basic_fields');
+    var Registry = require('web.field_registry');
+    var FormController = require('web.FormController');
+    var utils = require('web.utils');
+    var session = require('web.session');
+    var field_utils = require('web.field_utils');
+    var _t = core._t;
+    var QWeb = core.qweb;
+    const WIDTH = 500;
+    const HEIGHT = 376;
+
+    var FieldPainting = BasicFields.FieldBinaryImage.extend({
+        template: 'FieldPainting',
+        placeholder: "/web_widget_image_paint/static/src/img/image-placeholder.png",
+        events: _.extend({}, BasicFields.FieldBinaryImage.prototype.events, {
+            "click #reset": "reset_clicked",
+            "click #add_marker": "add_marker",
+            "click #drawing-mode": "drawing_mode",
+            "change #drawing-mode-selector": "drawing_mode_selection",
+            "change #drawing-color": "drawingColorEl_onchange",
+            "change #drawing-shadow-color": "drawingShadowColorEl_onchange",
+            "change #drawing-line-width": "drawingLineWidthEl_onchange",
+            "change #drawing-shadow-width": "drawingShadowWidth_onchange",
+            "change #drawing-shadow-offset": "drawingShadowOffset_onchange",
+            "change #file_paint": "file_paint_onchange",
+            "click #edit_click": "edit_click",
+            "click #clear_click": "reset_clicked",
+            'click .save_paint': 'save_paint',
+        }),
+
+        // **********************************************************
+        // * edit_click
+        // **********************************************************
+        edit_click: function (ev) {
```

Comment:
There is no need for this, the `<input>` takes care of it, plus right now the dialog opens twice.

Reply:
ok